src/main/java/org/broadinstitute/hellbender/utils/read/AlignmentUtils.java

```diff
@@ -196,6 +196,56 @@ public static GATKRead createReadAlignedToRef(final GATKRead originalRead,
         return Arrays.copyOfRange(bases, basesStart, basesStop + 1);
     }
 
+    public static byte[] getBasesAlignedOneToOne(final GATKRead read) {
+        return getSequenceAlignedOneToOne(read, true, (byte)'N');
+    }
+
+    public static byte[] getBaseQualsAlignedOneToOne(final GATKRead read) {
+        return getSequenceAlignedOneToOne(read, false, (byte)0);
+    }
+
+    public static byte[] getSequenceAlignedOneToOne(final GATKRead read, final boolean forBases, final byte padWith) {
```

Comment:
Using a boolean param to indicate whether this is an operation on quality scores or read bases smells a bit. What about provide a lambda to obtain the original-not-copied byte array given the read. 
```
public static byte[] getSequenceAlignedOneToOne(final GATKRead read, Function<GATKRead, byte[]> bytesProvider, int padWith) {
   ...
} 
```